homeassistant/components/nzbget/switch.py

```diff
@@ -0,0 +1,71 @@
+"""Support for NZBGet switches."""
+from typing import Callable, List, Optional
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_NAME
+from homeassistant.helpers.entity import Entity
+from homeassistant.helpers.typing import HomeAssistantType
+
+from . import NZBGetEntity
+from .const import DATA_COORDINATOR, DOMAIN
+from .coordinator import NZBGetDataUpdateCoordinator
+
+
+async def async_setup_entry(
+    hass: HomeAssistantType,
+    entry: ConfigEntry,
+    async_add_entities: Callable[[List[Entity], bool], None],
+) -> None:
+    """Set up NZBGet sensor based on a config entry."""
+    coordinator: NZBGetDataUpdateCoordinator = hass.data[DOMAIN][entry.entry_id][
+        DATA_COORDINATOR
+    ]
+
+    switches = [
+        NZBGetDownloadSwitch(
+            coordinator,
+            entry.entry_id,
+            entry.data[CONF_NAME],
+        ),
+    ]
+
+    async_add_entities(switches)
+
+
+class NZBGetDownloadSwitch(NZBGetEntity, Entity):
```

Comment:
You can't use the CoordinatorEntity here?

Reply:
NZBGetEntity is actually inheriting from such. we dont need the Entity subclass here

Reply:
Yes, I see it now. That's the trouble with the representation of the changes on this page, you can't see much of the existing code :)